hca/dss/util/__init__.py

```diff
@@ -58,7 +58,13 @@ def hardlink(source, link_name):
         if source_stat.st_dev != dest_stat.st_dev or source_stat.st_ino != dest_stat.st_ino:
             raise
     except OSError as e:
-        if e.errno == errno.EMLINK:
+        # See https://docs.python.org/3/library/errno.html
+        copy_on_error = [
+            errno.EMLINK,  # too many links
+            errno.EACCES,  # permission denied
```

Comment:
```suggestion
```

Why is EACCES included here? Our current philosophy is to catch only expected errors that we know can be addressed by falling back to a copy. I have seen no evidence that EACCES is to be expected and should be handled by falling back to a copy. Mere similarity isn't sufficient evidence for me. In fact, I suspect that it is very likely that if `os.link` fails with EACCES, so will `open(…, 'w')` which is what `shutil.copyfile` uses under the hood.

Reply:
This is a good point, removed EACCES so now it is just EMLINK and EPERM